managed/services/management/dbaas/kubernetes_server.go

```diff
@@ -341,159 +316,129 @@ func (k kubernetesServer) RegisterKubernetesCluster(ctx context.Context, req *db
+		operatorsToInstall["psmdb"] = true
+	}
 
-			installPlanName, err := getInstallPlanForSubscription(ctx, k.dbaasClient, req.KubeAuth.Kubeconfig, namespace, operator)
-			if err != nil {
-				k.l.Errorf("cannot get install plan for subscription %q: %s", operator, err)
-			}
+	settings, err := models.GetSettings(k.db.Querier)
+	if err != nil {
+		k.l.Errorf("cannot get PMM settings to start Victoria Metrics: %s", err)
+		return nil, errors.Wrap(err, "cannot get PMM settings to start Victoria Metrics")
+	}
 
-			if err := approveInstallPlan(ctx, k.dbaasClient, req.KubeAuth.Kubeconfig, namespace, installPlanName); err != nil {
-				k.l.Errorf("cannot approve the PXC install plan: %s", err)
-			}
+	if settings.PMMPublicAddress != "" {
+		var apiKeyID int64
+		var apiKey string
+		apiKeyName := fmt.Sprintf("pmm-vmagent-%s-%d", req.KubernetesClusterName, rand.Int63()) //nolint:gosec
+		apiKeyID, apiKey, err = k.grafanaClient.CreateAdminAPIKey(ctx, apiKeyName)
+		if err != nil {
+			k.l.Errorf("cannot create Grafana admin API key: %s", err)
+			return nil, errors.Wrap(err, "cannot create Grafana admin API key")
 		}
 
-		if psmdbOperatorVersion != nil && (clusterInfo.Operators == nil || clusterInfo.Operators.PsmdbOperatorVersion == "") {
-			operator := "percona-server-mongodb-operator"
-
-			if err := k.installOperator(ctx, operator, namespace, "percona-server-mongodb-operator.v1.11.0", "stable", req.KubeAuth.Kubeconfig); err != nil {
-				k.l.Errorf("cannot install PSMDB operator in the new cluster: %s", err)
-			}
-
-			installPlanName, err := getInstallPlanForSubscription(ctx, k.dbaasClient, req.KubeAuth.Kubeconfig, namespace, operator)
-			if err != nil {
-				k.l.Errorf("cannot get install plan for subscription %q: %s", operator, err)
-			}
+		pmmParams := &dbaascontrollerv1beta1.PMMParams{
+			PublicAddress: fmt.Sprintf("https://%s", settings.PMMPublicAddress),
+			Login:         "api_key",
+			Password:      apiKey,
+		}
 
-			if err := approveInstallPlan(ctx, k.dbaasClient, req.KubeAuth.Kubeconfig, namespace, installPlanName); err != nil {
-				k.l.Errorf("cannot approve the PSMDB install plan: %s", err)
+		_, err := k.dbaasClient.StartMonitoring(ctx, &dbaascontrollerv1beta1.StartMonitoringRequest{
+			KubeAuth: &dbaascontrollerv1beta1.KubeAuth{
+				Kubeconfig: req.KubeAuth.Kubeconfig,
+			},
+			Pmm: pmmParams,
+		})
+		if err != nil {
+			e := k.grafanaClient.DeleteAPIKeyByID(ctx, apiKeyID)
+			if e != nil {
+				k.l.Warnf("couldn't delete created API Key %v: %s", apiKeyID, e)
 			}
+			k.l.Errorf("couldn't start monitoring of the kubernetes cluster: %s", err)
+			return nil, errors.Wrap(err, "couldn't start monitoring of the kubernetes cluster")
 		}
+	}
 
-		if clusterInfo.Operators == nil || clusterInfo.Operators.OlmOperatorVersion == "" {
-			operator := "victoriametrics-operator"
+	// Install the operators in the background.
+	go k.installDefaultOperators(operatorsToInstall)
```

Comment:
We may have a situation when we run StartMonitoring while VM operator is not installed, don't we?

Reply:
You are right. Thanks.